library/src/cache/patch_manager.rs

```diff
@@ -213,7 +221,11 @@ impl PatchManager {
         }
 
         // If the last boot we tried was this patch, make sure we succeeded or the patch is bad.
-        if self.is_patch_last_attempted_patch(patch.number) {
+        // If we are currently in the process of booting this patch for the first time, this check
+        // will always fail, so skip it.
```

Comment:
Is the real fix to explicitly validate as a separate phase? If so, we should say that as a TODO or issue.

Reply:
Added a todo, this goes away in the next PR that's on deck and approved.